src/index.js

```diff
@@ -1,40 +1,71 @@
 import './style.css';
+import Features from './module/functionality.js';
+import Storage from './module/localstorage.js';
 
-const container = document.querySelector('.todo-listing');
-
-const todoList = [
-  {
-    description: 'Read my books',
-    completed: true,
-    index: 3,
-  },
-  {
-    description: 'Study data structures',
-    completed: true,
-    index: 1,
-  },
-  {
-    description: 'Learn new coding concepts',
-    completed: false,
-    index: 2,
-  },
-
-];
-
-const renderTask = () => {
-  const sortedList = todoList.sort((item1, item2) => item1.index - item2.index);
-
-  for (let i = 0; i < sortedList.length; i += 1) {
-    container.innerHTML += `
-    <div class="task-container">
-      <form class="task-form">
-        <input type="checkbox" class="checkbox">
-      </form>
-    <p class="task-description">${sortedList[i].description}</p>
-  </div>
-  <hr>
-    `;
+const ulContainer = document.querySelector('.list-content');
+
+const renderTasks = () => {
+  let taskList;
+
+  if (Storage.getLocalStorage() === null) {
+    taskList = [];
+  } else {
+    taskList = Storage.getLocalStorage();
   }
+
+  let content = '';
+
+  taskList.forEach((task, id) => {
+    content += `
+    <li class="list-items">
+    <div class="render-div">
+      <input class="check" type="checkbox">
+      <input class="task-description" id="task${id}" value=${task.description} />
+    </div>
+    <div class="icon-content">
+      <i id="removeTask${id}" class="sective fa-solid fa-trash-can"></i>
+    </div>
+  </li>`;
+  });
+
+  ulContainer.innerHTML = content;
+
+  taskList.forEach((task, index) => {
+    const removeTask = document.getElementById(`removeTask${task.index}`);
+    if (removeTask) {
+      removeTask.addEventListener('click', () => {
+        Features.removeTask(index);
+        removeTask.remove();
+        renderTasks();
+      });
+    }
+  });
+
+  taskList.forEach((tasking, index) => {
+    const updatedInput = document.getElementById(`task${tasking.index}`);
+    if (updatedInput) {
+      updatedInput.addEventListener(('keydown'), (e) => {
+        if (e.code === 'Enter') {
+          e.preventDefault();
+          Features.updateTask(updatedInput.value, index);
+          renderTasks();
+
+          updatedInput.value = '';
+        }
+      });
+    }
+  });
```

Comment:
- [ ] As pointed out by the previous code reviewer which is a [requirement ](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md)for this project, " `Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1).., By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5). ` ", please make sure when a new task is added to the list, the property index set to the value of the new array length, and deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. Please kindly fix that.